src/backend/tcop/postgres.c

```diff
@@ -1589,25 +1591,30 @@ send_guc_to_QE(List *guc_list, bool is_restore)
 		}
 		PG_CATCH();
 		{
-			/* if some guc can not restore successful
-			 * we can not keep alive gang anymore.
-			 */
-			DisconnectAndDestroyAllGangs(true);
```

Comment:
Why were these lines moved? In other similar places, the `DisconnectAndDestroyAllGangs` function is called specifically in the `PG_CATCH` block.